### PR TITLE
spline #1193 Admin: add `--dry-run` option

### LINKED ISSUE #1193: Admin: add `--dry-run` option

### PR DESCRIPTION
**Pull request by wajda**
fixes #1193

### CODE REVIEW

File: persistence/src/main/scala/za/co/absa/spline/persistence/DryRunnable.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    else code
+  }
+
+  def unlessDryRun[T <: AnyRef](code: => T): T = {
+    unlessDryRun(code, null).asInstanceOf[T]
+  }
+
+  def unlessDryRunAsync[T <: Future[U], U](code: => U): U = {
+    unlessDryRun(code, Future.successful(null)).asInstanceOf[U]
+  }
```
Comment (cerveada) [inline persistence/src/main/scala/za/co/absa/spline/persistence/DryRunnable.scala]: What is `T` needed for? Seems to be unused. Should some of the `U`s be `T`s?

File: admin/src/test/scala/za/co/absa/spline/admin/AdminCLISpec.scala [hunk 2]
```diff
@@ -123,6 +124,14 @@ class AdminCLISpec
...
     }
 
+    it should "when called with option --dry-run, set dryRun mode to true" in {
+      val output = captureStdOut {
+        cli.exec(Array("db-exec", "arangodbs://foo/bar", "--dry-run"))
+      }
+      dryRunModeCaptor.getValue should be(true)
+      output should include("Dry-run mode activated")
+    }
+
```
Comment (cerveada) [inline admin/src/test/scala/za/co/absa/spline/admin/AdminCLISpec.scala:134]: Some "integration" test validating that when dry run is set there is no interaction with the parts of the code that should not be called, could be also useful. Maybe in different filet though.

File: persistence/src/main/scala/za/co/absa/spline/persistence/DryRunnable.scala (hunk 1 above)
Comment (wajda) [reply persistence/src/main/scala/za/co/absa/spline/persistence/DryRunnable.scala]: Ah, that's a typo, thanks :)

File: admin/src/test/scala/za/co/absa/spline/admin/AdminCLISpec.scala (hunk 2 above)
Comment (wajda) [reply admin/src/test/scala/za/co/absa/spline/admin/AdminCLISpec.scala:134]: I thought about it, but it has to be an integration test with a real ArangoDB. Unfortunately capturing this behavior with mocks is impossible. Due to the nature of the ArangoDB driver there is no one place where you can intercept all write operations, or isolate all write method calls. The guard conditions are spread almost freely over the Admin codebase.
I'll see what I can do.

Comment (wajda) [reply admin/src/test/scala/za/co/absa/spline/admin/AdminCLISpec.scala:134]: added ArangoDb test container, fixtures and a couple of integration tests

**Review by cerveada — APPROVED** [review]

**Pull request merged**